test/unit/services/auth_service_test.dart

```diff
@@ -147,5 +156,125 @@ void main() {
+            emitsDone,
+          ]));
+    });
+
+    test(
+        'emailSignInStream catches firebaseAuthExceptions and emits StoreProblem actions',
+        () async {
+      // init service to throw firebaseAuthException on sign in
+      final service =
+          AuthService(FakeFirebaseAuthSignInException(), null, null);
+
+      /// Check that authService emits [AddProblem] action with info
+      /// from caught [firebaseAuthException].
+      /// We use a [TypeMatcher] as it's difficult to create the expected
+      /// [Problem] due to the [Problem.trace] member
+      expect(
+          service.emailSignInStream('test@email.com', 'test_password'),
+          emitsInOrder(<dynamic>[
+            UpdateEmailAuthOptionsPage(step: AuthStep.signingInWithEmail),
+            UpdateEmailAuthOptionsPage(step: AuthStep.waitingForInput),
+            TypeMatcher<AddProblem>()
+              ..having((p) => p.problem.type, 'type', ProblemType.emailSignIn)
+              ..having(
+                  (p) => p.problem.info,
+                  'info',
+                  BuiltMap<String, Object>(
+                      {'code': 'test error: cant find user'}))
+              ..having((p) => p.problem.message, 'message',
+                  equals('firebase auth exception error')),
+            emitsDone
+          ]));
+    });
+
+    test('emailSignInStream catches errors and emits StoreProblem actions',
+        () async {
+      // Init service that throws exception when signing in with firebase
+      final service = AuthService(FakeFirebaseAuthThrows(), null, null);
+
+      /// Check that emailSignInStream emits addProblem action with info from caught error.
+      /// We use a [TypeMatcher] as it's difficult to create the expected
+      /// [Problem] due to the [Problem.trace] member
+      expect(
+          service.emailSignInStream('test@email.com', 'test_password'),
+          emitsInOrder(<dynamic>[
+            UpdateEmailAuthOptionsPage(step: AuthStep.signingInWithEmail),
+            UpdateEmailAuthOptionsPage(step: AuthStep.waitingForInput),
+            TypeMatcher<AddProblem>()
+              ..having((p) => p.problem.type, 'type', ProblemType.emailSignIn)
+              ..having((p) => p.problem.message, 'message',
+                  equals('firebase auth exception error')),
+            emitsDone
+          ]));
+    });
+
+    test('signOut signs out of all providers', () async {
+      // init all auth providers
+      final mockFireBaseAuth = MockFireBaseAuth();
+      final mockGoogleSignIn = MockGoogleSignIn();
+      final mockAppleSignIn = MockAppleSignIn();
+      final authService =
+          AuthService(mockFireBaseAuth, mockGoogleSignIn, mockAppleSignIn);
+
+      await authService.signOut();
+
+      verify(mockFireBaseAuth.signOut());
+      verify(mockGoogleSignIn.signOut());
+      // TODO: add apple sign in when #232 is completed. https://github.com/crowdleague/crowdleague/issues/232
```

Comment:
grab me for a chat about this if you don't mind - I think I'd like to get use away from allowing TODOs in the dev branch

Reply:
I made [Style Guide > avoid-pushing-todos-to-dev](https://github.com/crowdleague/crowdleague/wiki/Style-Guide-(WIP)#avoid-pushing-todos-to-dev) and will add a note to discuss on Monday 🙂

Reply:
I've added an item to Monday standup